src/main/Backend.java

```diff
@@ -7,141 +7,80 @@
-                        // We will not add the object and wait for the next message which should be init
-                        myOtpMbox.send(last_pid, ERLObjectId);
-                        System.out.println("The object is still in the pipeline so we have to wait to get it");
-                    }
-                } else {
-                    CRDT crdt_object = ObjectTable.get(ERLObjectId);
-                    switch (status) {
-                    case "read":
-                        System.out.println("Doing read call");
-                        myOtpMbox.send(last_pid, new OtpErlangBinary(crdt_object.read()));
-                        break;
-
-                    case "invoke":
-                        System.out.println("Doing invoke call");
-                        try {
-                            GenericFunction func = (GenericFunction) binary.getObject();
-                            System.out.println("Doing invoke");
-                            crdt_object.invoke(func);
-                        } catch (ClassCastException e) {
-                            System.out.println("Did this already");
-                        }
-                        myOtpMbox.send(last_pid, ERLObjectId);
-                        break;
-
-                    case "snapshot":
-                        System.out.println("Doing antidote snapshot update");
-                        crdt_object.snapshot();
-                        byte[] b = new byte[20];
-                        new Random().nextBytes(b);
-                        OtpErlangBinary new_key = new OtpErlangBinary(b);
-                        //ObjectTable.remove(ERLObjectId);
-                        ObjectTable.put(new_key, crdt_object);
-                        OtpErlangObject[] emptypayload = new OtpErlangObject[2];
-                        emptypayload[0] = new_key;
-                        emptypayload[1] = new OtpErlangBinary(crdt_object);
-                        OtpErlangTuple new_snapshot = new OtpErlangTuple(emptypayload);
-                        myOtpMbox.send(last_pid, new_snapshot);
-                    }
-                }
-            } catch (Exception e) {
-                e.printStackTrace();
-                // return some failure
-                OtpErlangString errormsg = new OtpErlangString("hi");
-                // myOtpMbox.send(last_pid, errormsg);
-                OtpErlangAtom atom = new OtpErlangAtom("error");
-                myOtpMbox.send(last_pid, atom);
+                // assert that object is in table else request it
+                CRDT crdt_object = ObjectTable.get(JavaObjectId);
+                GenericFunction func = (GenericFunction) binary.getObject();
+                crdt_object.invoke(func);
+            } catch (ClassCastException e) {
+                // Check that we currently have a valid object in the table
+                System.out.println("Did this already");
```

Comment:
Another quick assert would be good, and I'm always a fan of `/*intentionally ignore this exception*/`

Reply:
// Commented